### ISSUE TITLE
I have problems using Cart

### ISSUE THREAD

**Issue by ngominhtrungit** [RayOkaah/woocommerce#35]
I have problems using Cart. Can someone give 1 example to use myCart(WooCart) and WooOrderPayload?
Thanks.

**Comment by CarloDotlog** [comment]
Hi, me too. My attempts never succeeded. So I've skipped the problem and I have adopted the [CoCart plugin ](https://docs.cocart.xyz/index.html#introduction) with this [add-on](https://docs.cocart.xyz/index.html#cocart-add-ons-get-cart-enhanced) enabled.
I've made implementation at the app level, so I've no changed the library. 
I hope this can help. 
Happy coding.

**Comment by ngominhtrungit** [comment]
Only cart use cocart? Because now i use my phone, I don't search

**Comment by CarloDotlog** [comment]
Yes, I use CoCart only to manage the cart. So I can add/remove items to the cart and every change is saved on WordPress DB. 
You can continue to use this library and add your code to implement CoCart.

**Comment by ngominhtrungit** [comment]
Yeah! I'm try it, But I will try to code using WooCart of the woocommercer package.  thank you.

**Comment by ngominhtrungit** [comment]
> Yes, I use CoCart only to manage the cart. So I can add/remove items to the cart and every change is saved on WordPress DB.
> You can continue to use this library and add your code to implement CoCart.

**Comment by ngominhtrungit** [comment]
> Có, tôi chỉ sử dụng CoCart để quản lý giỏ hàng. Vì vậy, tôi có thể thêm / bớt các mặt hàng vào giỏ hàng và mọi thay đổi đều được lưu trên WordPress DB.
> Bạn có thể tiếp tục sử dụng thư viện này và thêm mã của mình để triển khai CoCart.

hello, do you have documentation on flutter with cocart plugin?

**Comment by ngominhtrungit** [comment]
> Yes, I use CoCart only to manage the cart. So I can add/remove items to the cart and every change is saved on WordPress DB.
> You can continue to use this library and add your code to implement CoCart.

https://github.com/RayOkaah/woocommerce/issues/35

**Comment by ngominhtrungit** [comment]
> Yes, I use CoCart only to manage the cart. So I can add/remove items to the cart and every change is saved on WordPress DB.
> You can continue to use this library and add your code to implement CoCart.

hello, do you have documentation on flutter with cocart plugin?

**Comment by CarloDotlog** [comment]
Hi! No, there is no Flutter documentation. 
You can find the plugin documentation [here ](https://docs.cocart.xyz/index.html). You need to create the CoCart entities classes and then call the CoCart endpoints. As I said, I suggest using this [add-on](https://docs.cocart.xyz/index.html#cocart-add-ons-get-cart-enhanced).
Happy conding!

**Comment by ngominhtrungit** [comment]
Thank you.i use it. Thank for share. If i has issue. Pls help thank

**Comment by federico2390** [comment]
@CarloDotLog Do u have an example of api for adding item in cart with cocart?

**Comment by CarloDotLog** [comment]
Yes I have something.. somewhere... I use this package into an old project dated 1+ year. 
I'll update you as soon as I find the code 😉

**Comment by federico2390** [comment]
@CarloDotLog Yes please if u find the code can share here or via mail?
Thx for help

**Comment by CarloDotLog** [comment]
@federico2390 As I said, this code is 15 months old.. I think you must update it. =/ 
If you need help, we could see it together. 
--

For the entities, I have a Gist --> https://gist.github.com/CarloDotLog/d85d24e6e23e2f019386c5f737d1216d

The following method add an Item or Variation to CoCart. 
It's based on v1 CoCart API and woocommerce: ^0.9.7.

```
static const String URL_COCART_API_PATH = "/wp-json/cocart/v1/";

/// Accepts an int [productId] of a product, int quantity, int [variationId] of a product or product variation
/// Related endpoint : cocart/v1/add-item
///
Future<Items> addItemToCoCart({required String cartKey, required int productId, required int quantity, int? variationId}) async {
  Map<String, dynamic> data = {
    'cart_key': cartKey,
    'product_id': "$productId",
    'quantity' : "$quantity",
  };
  if(variationId!=null) data['variation_id'] = "$variationId";
  debugPrint("$data");
  
  String authToken = await wooCommerce.getAuthTokenFromDb();
  _urlHeader['Authorization'] = 'Bearer $authToken';
  
  final response = await http.post(Uri.parse(baseUrl + URL_COCART_API_PATH + "add-item"), headers: _urlHeader, body: data);
  debugPrint(baseUrl + URL_COCART_API_PATH + 'add-item');

  if (response.statusCode >= 200 && response.statusCode < 300) {
    final jsonStr = json.decode(response.body);

    debugPrint("added to my cart : "+jsonStr.toString());
    return Items.fromJson(jsonStr);
  } else {
    WooCommerceError err = WooCommerceError.fromJson(json.decode(response.body));
    throw err;
  }
}
```
